### PR TITLE
Feature/mongo db

### PR DESCRIPTION
**Pull request by Roland-Hufnagel**

### CODE REVIEW

File: components/CreateSurveyForm.js [hunk 1]
```diff
@@ -55,7 +52,12 @@ export default function CreateSurvey({ title, questions }) {
   }
   function handleAdd() {
     const newQuestions = [...survey.questions];
-    newQuestions.push({ title: "", type: "" });
+    newQuestions.push({
+      title: "",
+      type: "",
+      id: Math.random().toString(),
```
Comment (Kheirah) [inline components/CreateSurveyForm.js:58]: You're better off using nanoid for generating unique ids. Math.random isn't inappropriate here though, but usuuuuaally you'd expect a sequence of characters and/or numbers for ids and not just numbers.

To sum up, Math.random works. But it appeals more to convention if you used something like nanoid.

File: pages/api/edit/[id].js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    const survey = await SurveyModel.findByIdAndDelete(id);
+    return response
+      .status(200)
+      .json({ message: "Survey deleted", deletedId: id });
+  } else if (request.method === "PATCH") {
+    const postData = JSON.parse(request.body);
+    const newSurvey = await SurveyModel.findByIdAndUpdate(id, {
+      title: postData.title,
+      questions: postData.questions,
+    });
```
Comment (Kheirah) [inline pages/api/edit/[id].js:17]: Currently, you're allowing only two keys be updated even if the body of your request contained more data.
What if there's more to update than just the title and questions, sometime in the future? I mean, why not pass the entire postData?

File: pages/create/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  return (
+    <CreateSurveyForm
+      title=""
+      description=""
+      date={new Date()}
+      url={`${host}/survey/`}
+      questions={[
+        {
+          id: Math.random().toString(),
```
Comment (Kheirah) [inline pages/create/index.js:34]: Same thing here I mentioned before: you could use nanoid to comply with general expectation.

File: pages/survey/[id].js [hunk 4]
```diff
@@ -1,34 +1,54 @@
...
+    for (const key in data) {
+      questions.find((question) => question.id === key).answers.push(data[key]);
+    }
+    try {
+      const response = await fetch(`/api/subscribe/${id}`, {
+        method: "PUT",
+        body: JSON.stringify(questions),
+      });
+      const result = await response.json();
+      console.log(result);
```
Comment (Kheirah) [inline pages/survey/[id].js:33]: Consider removing logs when you no longer need them

**Review by Kheirah — APPROVED** [review]
Really good work!

I proposed some minor changes, nothing wild.

Keep it up! =)

File: pages/api/edit/[id].js (hunk 2 above)
Comment (Roland-Hufnagel) [reply pages/api/edit/[id].js:17]: you are absolute right!!

**Pull request merged**